provision/pool/pool.go

```diff
@@ -53,12 +60,28 @@ type AddPoolOptions struct {
 	Default     bool
 	Force       bool
 	Provisioner string
+
+	Labels map[string]string
 }
 
 type UpdatePoolOptions struct {
 	Default *bool
 	Public  *bool
 	Force   bool
+
+	Labels map[string]string
+}
+
+func (p *Pool) GetAffinity() (*apiv1.Affinity, error) {
+	if affinity, ok := p.Labels[affinityKey]; ok {
+		var k8sAffinity apiv1.Affinity
+		if err := json.Unmarshal([]byte(affinity), &k8sAffinity); err != nil {
```

Comment:
@arthurcgc: nice to have: also support `yaml.Unmarshal` first and if an error tries json.
@pedrokiefer what do you think about ?

Reply:
I'm not sure. Maybe only yaml?The question is how we want to expose this in the terraform provider? Yaml is more kubernetes like.

Reply:
I think only one mode (yaml or json) should suffice... If we're thinking of using it with terraform we should maintain the pattern we're implementing in the terraform-provider, which I'm not sure what it is xD